pkg/config/analysis/local/analyze_test.go

```diff
@@ -162,6 +169,85 @@ func TestAddRunningKubeSource(t *testing.T) {
 	g.Expect(sa.stores).To(HaveLen(1))
 }
 
+func TestAnalyseInClusterSource(t *testing.T) {
+	g := NewWithT(t)
+	mk := kube.NewFakeClient()
+	name := "name"
+	namespace := "ns"
+	cancel := make(chan struct{})
+	for _, schema := range []collection.Schema{
+		collections.IstioNetworkingV1Alpha3Virtualservices,
+		collections.IstioNetworkingV1Alpha3Serviceentries,
+	} {
+		crdclient.CreateCRD(t, mk, schema.Resource())
+	}
+	vs := &clientv1alpha3.VirtualService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha3.VirtualService{
+			Http: []*v1alpha3.HTTPRoute{
+				{
+					Route: []*v1alpha3.HTTPRouteDestination{
+						{
+							Destination: &v1alpha3.Destination{
+								Host: "none.default.svc.cluster.local",
+								Port: &v1alpha3.PortSelector{
+									Number: 8080,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	if _, err := mk.Istio().NetworkingV1alpha3().VirtualServices(namespace).Create(context.TODO(), vs, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Error creating virtual service: %v", err)
+	}
+	sa := NewSourceAnalyzer(analysis.Combine("vs", &vsanalyzer.DestinationHostAnalyzer{}), resource.Namespace(namespace), "", nil, true, timeout)
+	sa.AddRunningKubeSource(mk)
+	result, err := sa.Analyze(cancel)
+	if err != nil {
+		t.Fatal(err)
+	}
+	line, err := locateErrorLine(mk, namespace, name, "host")
```

Comment:
why not just write out the string? Otherwise we don't know if the test code is correct or not